### PR TITLE
refactor: use random service name in tests

### PR DESCRIPTION
**Pull request by rebeccabrennan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-245

### Change description ###

Starting to randomise test suite. Resource definition able to take randomised values in test suite and service name initially randomised.

### CODE REVIEW

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 1]
```diff
@@ -37,39 +37,46 @@
...
         resourceId = UUID.randomUUID().toString();
         accessorId = UUID.randomUUID().toString();
+        serviceName = UUID.randomUUID().toString();
+        importerService.addService(serviceName);
+        importerService.addResourceDefinition(ResourceDefinition.builder()
+            .serviceName(serviceName)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .build());
+        resourceDefinition = createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME);
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: Why resource definition is created twice? Once by `ResourceDefinition.builder()`directly and second time by `createResourceDefinition` method? By the way the fact I asked that question also means I do not like it as is ;)

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 2]
```diff
@@ -37,39 +37,46 @@
...
     private static final String PARENT_ATTRIBUTE = "/parent";
     private static final String CHILD_ATTRIBUTE = "/parent/child";
-    private static final ResourceDefinition RESOURCE_DEFINITION =
-        createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+    private static ResourceDefinition resourceDefinition;
     private static AccessManagementService service = initService(AccessManagementService.class);
     private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
     private String resourceId;
     private String accessorId;
+    String serviceName;
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: Why there is no visibility modifier? If no good reason then specify.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java [hunk 3]
```diff
@@ -32,54 +33,55 @@
...
 import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
 
-@SuppressWarnings("PMD.TooManyMethods")
+@SuppressWarnings({"PMD.TooManyMethods", "LineLength"})
 class GetResourceDefinitionsWithCreatePermissionIntegrationTest extends PreconfiguredIntegrationBaseTest {
     private static AccessManagementService service = initService(AccessManagementService.class);
     private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
 
-    private final ResourceDefinition resource = createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+    private static ResourceDefinition resourceDefinition;
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: Why is that static? Any good reason? If not make is non static.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java [hunk 4]
```diff
@@ -32,54 +33,55 @@
...
     private final ResourceDefinition otherResource = createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE,
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: Other reassume still uses fixed service name. Change it.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java [hunk 5]
```diff
@@ -25,89 +27,97 @@
...
     }
 
     @Test
     void shouldAddNewEntryIntoDatabaseWhenUniqueEntry() {
         service.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
-        service.grantDefaultPermission(createDefaultPermissionGrant(ImmutableSet.of(READ)));
+        service.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ImmutableSet.of(READ)));
 
-        assertThat(databaseHelper.countDefaultPermissions(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.countDefaultPermissions(serviceName, RESOURCE_TYPE, RESOURCE_NAME,
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java]: Can you refactor `databaseHelper.countDefaultPermissions` to take instance of `ResourceDefinition`?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java [hunk 6]
```diff
@@ -25,89 +27,97 @@
...
             ROOT_ATTRIBUTE.toString(), ROLE_NAME, Permissions.sumOf(ImmutableSet.of(READ)))).isEqualTo(1);
 
-        assertThat(databaseHelper.getResourceAttribute(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME,
+        assertThat(databaseHelper.getResourceAttribute(serviceName, RESOURCE_TYPE, RESOURCE_NAME,
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java]: Can you refactor `databaseHelper.getResourceAttribute` as well to work with ResourceDefinition's?

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java [hunk 7]
```diff
@@ -30,23 +27,19 @@ private DefaultRoleSetupDataFactory() {
...
 
-    public static DefaultPermissionGrant createDefaultPermissionGrant(Set<Permission> permissions) {
-        return createDefaultPermissionGrant(JsonPointer.valueOf(""), permissions, PUBLIC);
+    public static DefaultPermissionGrant createDefaultPermissionGrant(ResourceDefinition resourceDefinition, Set<Permission> permissions) {
+        return createDefaultPermissionGrant(JsonPointer.valueOf(""), permissions, resourceDefinition, PUBLIC);
     }
 
-    public static DefaultPermissionGrant createDefaultPermissionGrant(JsonPointer attribute, Set<Permission> permissions) {
-        return createDefaultPermissionGrant(attribute, permissions, PUBLIC);
+    public static DefaultPermissionGrant createDefaultPermissionGrant(JsonPointer attribute, ResourceDefinition resourceDefinition, Set<Permission> permissions) {
```
Comment (geek-soft) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java]: Please keep consistent order. That means resource definition comes before attribute and permissions. Same in other places.

**Comment by geek-soft** [comment]
Remove dot from the PR title please 😠 

Also it would be much simpler if you said `refactor: use random service name in tests`. Way more readable IMO.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 1 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 2 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java (hunk 3 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java (hunk 4 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java (hunk 7 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java (hunk 5 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java (hunk 4 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 8]
```diff
@@ -37,39 +37,42 @@
...
+    private ResourceDefinition resourceDefinition;
 
     @BeforeEach
     void setUp() {
         resourceId = UUID.randomUUID().toString();
         accessorId = UUID.randomUUID().toString();
+        serviceName = UUID.randomUUID().toString();
+        importerService.addService(serviceName);
+        importerService.addResourceDefinition(
+            resourceDefinition = createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: Hacky, nice ;)

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java [hunk 9]
```diff
@@ -30,56 +31,51 @@
...
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.SingularField", "LineLength"})
 class GetResourceDefinitionsWithCreatePermissionIntegrationTest extends PreconfiguredIntegrationBaseTest {
     private static AccessManagementService service = initService(AccessManagementService.class);
     private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
-
-    private final ResourceDefinition resource = createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
-    private final ResourceDefinition otherResource = createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE,
-        RESOURCE_NAME + 2);
+    private ResourceDefinition resourceDefinition;
+    private ResourceDefinition otherResource;
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: It is not resource, it is resource. definition. Please correct name.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 10]
```diff
@@ -16,30 +16,37 @@
...
     private static AccessManagementService service = initService(AccessManagementService.class);
     private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
+    private String serviceName;
+    private ResourceDefinition resourceDefinition;
+
 
     @BeforeEach
     void setUp() {
+        serviceName = UUID.randomUUID().toString();
+        importerService.addService(serviceName);
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: Interestingly we could do `importerService.addService(serviceName = UUID.randomUUID().toString())`. I'd be keen to have it.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java [hunk 11]
```diff
@@ -8,32 +8,44 @@
...
 import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.OTHER_ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
 import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
 import static uk.gov.hmcts.reform.amlib.helpers.TestDataFactory.createGrant;
 import static uk.gov.hmcts.reform.amlib.helpers.TestDataFactory.createMetadata;
 import static uk.gov.hmcts.reform.amlib.helpers.TestDataFactory.createPermissions;
 
-@SuppressWarnings({"PMD.TooManyMethods", "PMD.AvoidDuplicateLiterals"})
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.SingularField", "PMD.AvoidDuplicateLiterals", "LineLength"})
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java]: What violates `PMD.SingularField` rule? Please give us an example.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java [hunk 12]
```diff
@@ -25,89 +27,93 @@
...
+        assertThat(databaseHelper.getResourceAttribute(resourceDefinition,
             ROOT_ATTRIBUTE.toString(), PUBLIC)).isNotNull();
     }
 
     @Test
     void shouldRemoveAllEntriesFromTablesWhenValuesExist() {
         service.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
-        service.addResourceDefinition(createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME));
-        service.addResourceDefinition(createResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME + 2));
+        service.addResourceDefinition(createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java]: Why `resourceDefinition` field is not used here?

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java (hunk 7 above)
Comment (geek-soft) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/DefaultRoleSetupDataFactory.java]: Not addressed :(

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 8 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 10 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java (hunk 9 above)
Comment (geek-soft) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java]: I mean rename `otherResource` to `otherResourceDefinition`

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java (hunk 11 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java]: 👍 Taking out now due to moving service name in to preconfiguredintegration test. Service name was only being used once and violated the pmd singular field rule.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java (hunk 12 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java]: 👍

**Pull request merged**